tutorials/week1-typescript.md

```diff
@@ -0,0 +1,1058 @@
+    *   [Optional and Default Parameters](#optional-and-default-parameters)
+    *   [Rest Parameters](#rest-parameters)
+    *   [Functions as Constructors](#functions-as-constructors)
+    *   [Arrow Functions](#arrow-functions)
+    *   [Function Overloads](#function-overloads)
+*   [Classes](#classes)
+    *   [Creating a class](#creating-an-abstract-class)
+    *   [Creating an abstract class](#creating-an-abstract-class)
+*   [Type Aliases](#type-aliases)
+*   [Interfaces](#interfaces)
+*   [Custom types](#custom-types)
+*   [Generics](#generics)
+*   [Modules](#modules)
+*   [Object oriented programming concepts using typescript](#object-oriented-programming-concepts-using-typescript)
+    *   [Inheritance](#inheritance)
+    *   [Polymorphism](#polymorphism)
+    *   [Abstraction](#abstraction)
+    *   [Encapsulation](#encapsulation)
+*   [General Guidelines](#general-guidelines)
+
+Typescript is a superset of JavaScript. Thus, all JavaScript code is valid Typescript code. Typescript introduces concepts of optional typing, modules, and few additional features to Javascript. Let us begin by extending JavaScript to use the features of Typescript and primitive types.
+
+## Types in typescript
+
+### Boolean
+
+The boolean type has two values: true and false. 
+
+Example:
+```ts
+let a = true                // boolean
+     var b = false               // boolean
+     const c = true              // true
+     let d: boolean = true       // boolean
+     let e: true = true          // true
+```
+*   _Usage_: 
+You can compare them (with ==, ===, ||, &&, and ?), negate them (with !).
+
+### Number
+
+Number is the set of all numbers: integers, floats, positives, negatives, Infinity, NaN. 
+
+Example: 
+```ts
+var b = Infinity * 0.10     // number
+const c = 5678              // 5678
+let d = a < b               // boolean
+let e: number = 100         // number
+let f: 26.218 = 26.218
+```
+
+*   _Usage_: 
+Numbers can do things, like addition (+), subtraction (-), modulo (%), and comparison (<). When working with long numbers, use numeric separators to make those numbers easier to read. 
+
+### BigInt
+
+The bigint type is the set of all BigInts, and supports things like addition (+), subtraction (-), multiplication (*), division (/), and comparison (<). 
+
+Example: 
+```ts
+let a = 1234n               // bigint
+const b = 5678n             // 5678n
+var c = a + b               // bigint
+let d = a < 1235            // boolean
+let e = 88.5n               // Error TS1353: A bigint literal must be an integer.
+let f: bigint = 100n        // bigint
+let g: 100n = 100n
+```
+
+*   _Usage_: 
+It lets you work with large integers without running into rounding errors. While the number type can only represent whole numbers it can hold numbers larger than 2<sup>53</sup> – 1. 
+
+### String
+
+String is the set of all strings and the things you can do with them like concatenate (+), slice (.slice),etc. 
+
+Example: 
```

Comment:
Examples for "any" were provided here instead of a string

Reply:
Do you want to add a commit to this branch correcting this mistake? Let me know, else I can take it up